### DISCUSSION TITLE
New clients aren't inserted in the right position on HiDPi screen and bsp layout

### DISCUSSION THREAD

**Discussion by dannycolin** [qtile/qtile#3516] [Q&A] [answered]
Hi,

When I open a second application on a workspace, it is placed under the first application instead of on the right. I use the bsp layout and tried to play with both `fair` and `lower_right` properties without any success. Also, I'm on a HiDPi 3:2 screen if that may help. And finally, the same configuration works fine on my external screen on the same machine and also on my other laptop. So that's why I think it may be related to an issue between Qtile and the HiDPi screen.

Thanks,

`~/.bash_profile`:
```
# Set DPI for Gtk and Qt applications
export GDK_DPI_SCALE=0.5
export GDK_SCALE=2
export QT_AUTO_SCREEN_SET_FACTOR=0
export QT_SCALE_FACTOR=1.75
export QT_FONT_DPI=96
```
`~/.Xresources`:
```
! Global configuration
Xcursor.size: 48
Xft.dpi: 192
```
`~/.config/qtile/config.py`:
```
layouts = [
    layout.Bsp(
        border_focus="#15539e",
        border_normal="#353535",
        border_width=2,
        grow_amount=5,
    ),
...
```

Good:
![bsp-good](https://user-images.githubusercontent.com/7339076/166327221-d973beed-cc86-421d-ad21-0517fe6279e5.gif)

Bad:
![bsp-bad](https://user-images.githubusercontent.com/7339076/166327301-8ef75100-41d9-4081-97f4-cd451c3c5bab.gif)

**Comment by elParaguayo** [comment, marked as answer]
Sorry for the delay.

Once you've opened a couple of windows can you run the following commands:
`qtile cmd-obj -o layout -f eval -a "self.root.w"`, `qtile cmd-obj -o layout -f eval -a "self.root.h"` and `qtile cmd-obj -o layout -f eval -a "self.ratio"`. Post the values here.

**Reply by dannycolin** [reply]
Width: (True, '3001')
Height: (True, '1953')
Ratio: (True, '1.6')

**Reply by elParaguayo** [reply]
Can you do it on both screens. One where it splits horizontally and the other where it splits vertically.

**Reply by elParaguayo** [reply]
The screen splits horizontally where the width is greater than the height * the ratio. So, for the numbers above, height (1953) * 1.6 > width and you get a vertical split.

**Reply by dannycolin** [reply]
3:2 2K:
Width: (True, '3001')
Height: (True, '1953')
Ratio: (True, '1.6')

16:9 1080p:
(True, '3840')
(True, '2112')
(True, '1.6')

**Reply by dannycolin** [reply]
Could it be that Qtile get confused because of the fractional scaling. To achieve that under X, you have to scale to 200% all the applications then zoom out the whole screen to get the equivalent of a 150% scaling.

**Reply by elParaguayo** [reply]
Well you've got different aspect ratios on those two screens. For the second, width (3840) > height (2112) * 1.6 so you do have a horizontal split there.

If you're scaling, aren't you scaling the same about horizontally and vertically? If so, it shouldn't make a difference to the above calculation.

If you want the same behavior, you need to change the `ratio` for the layout.

**Reply by dannycolin** [reply]
Oh TIL about screen ratio calculation! So setting the ratio to something like `1.0` should solve the issue right?

**Reply by elParaguayo** [reply]
It's the layout's ratio, not the screen's. It's the value that it looks at to decide when to split horizontally. 

But, yes, if the screen is narrower then you would want a lower ratio.

**Reply by dannycolin** [reply]
Ah! `1.5` is enough since that's the ratio of a 3:2. Oh so how come Qtile doesn't automatically detect my screen ratio and set it as the default instead of using `1.6` for all screens? Would it be a good idea to file a feature request for that?

Also, a huge thanks for your help!

**Reply by elParaguayo** [reply]
Interesting idea.

The `split_horizontal` value is only calculated at the time the node is added so you'd think it would be enough just to look at the screen's dimensions at that time to determine the correct ratio.

**Reply by dannycolin** [reply]
I don't know Qtile codebase so I can't help with that part sadly. I think I'm going to file a feature request for that based on our discussion here. :)

**Reply by elParaguayo** [reply]
That's fine. Feature requests are always welcome.
